src/command_modules/azure-cli-billing/azure/cli/command_modules/billing/custom.py

```diff
@@ -0,0 +1,20 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+
+def cli_billing_list_invoices(client, generate_url=False):
+    '''List all available invoices of the subscription'''
+    invoices = client.list(expand='downloadUrl' if generate_url else None)
+    return list(invoices)
```

Comment:
You could actually eliminate this custom command and just do reflection on the sdk.  You would have an entry like this in your params.py file:

`register_cli_argument('billing', 'expand', options_list=['--generate-url'], action='store_const', const='downloadUrl', default=False)`

However, what you have is perfectly fine.